shell/platform/tizen/external_texture_surface_gl_tizen.cc

```diff
@@ -105,15 +125,60 @@ bool ExternalTextureSurfaceGL::PopulateTexture(
   PFNEGLCREATEIMAGEKHRPROC n_eglCreateImageKHR =
       reinterpret_cast<PFNEGLCREATEIMAGEKHRPROC>(
           eglGetProcAddress("eglCreateImageKHR"));
-  const EGLint attribs[] = {EGL_IMAGE_PRESERVED_KHR, EGL_TRUE, EGL_NONE,
-                            EGL_NONE};
-  EGLImageKHR egl_src_image =
-      n_eglCreateImageKHR(eglGetCurrentDisplay(), EGL_NO_CONTEXT,
-                          EGL_NATIVE_SURFACE_TIZEN, tbm_surface, attribs);
+  EGLImageKHR egl_src_image = nullptr;
+
+  if (state_->gl_extention == kExternalTextureExtensionTypeNativeSurface) {
+    const EGLint attribs[] = {EGL_IMAGE_PRESERVED_KHR, EGL_TRUE, EGL_NONE,
+                              EGL_NONE};
+    egl_src_image =
+        n_eglCreateImageKHR(eglGetCurrentDisplay(), EGL_NO_CONTEXT,
+                            EGL_NATIVE_SURFACE_TIZEN, tbm_surface, attribs);
+  } else if (state_->gl_extention == kExternalTextureExtensionTypeDmaBuffer) {
+    EGLint attribs[50];
```

Comment:
I think this block is too long to read, how about adding new method?

Reply:
I thought it at first, but it wasn't used anywhere else, so I decided to leave it alone. It may be different depending on the personal readability view.

Reply:
Yes, I agree with you. nevertheless, it's worth thinking about again(https://refactoring.guru/extract-method)